### PR TITLE
fix(context): keep response cookies in newResponse instead of overwriting

### PR DESCRIPTION
**Pull request by ItsWendell**
Potential fix for https://github.com/honojs/hono/issues/2554

### Author should do the followings, if applicable

- [x] Add tests
- [x] Run tests
- [x] `yarn denoify` to generate files for Deno

### CODE REVIEW

File: src/context.test.ts [hunk 1]
```diff
@@ -244,6 +245,19 @@ describe('Context header', () => {
...
+
+  it('Should keep previous cookies in response headers', () => {
+    c.res.headers.append('set-cookie', 'foo=bar; Path=/')
+    setCookie(c, 'foo2', 'bar2', { path: '/' })
+
+    const res = c.json({ message: 'Hello' })
+
+    const cookies = res.headers.getSetCookie()
+
+    console.debug('test cookies', cookies)
```
Comment (yusukebe) [inline src/context.test.ts:257]: This line is not needed.

File: src/context.test.ts [hunk 2]
```diff
@@ -244,6 +245,19 @@ describe('Context header', () => {
...
+
+    expect(cookies.length).toBe(2)
```
Comment (yusukebe) [inline src/context.test.ts:259]: It's better to add tests to check the values as you wrote in your issue:

```ts
expect(cookies.includes('foo=bar; Path=/')).toBe(true)
expect(cookies.includes('foo2=bar2; Path=/')).toBe(true)
```

File: src/context.test.ts [hunk 3]
```diff
@@ -244,6 +245,19 @@ describe('Context header', () => {
     c.res = res
     expect(c.res.headers.getSetCookie().length).toBe(2)
   })
+
+  it('Should keep previous cookies in response headers', () => {
+    c.res.headers.append('set-cookie', 'foo=bar; Path=/')
+    setCookie(c, 'foo2', 'bar2', { path: '/' })
+
```
Comment (yusukebe) [inline src/context.test.ts:252]: You have to remove these break likes.

**Comment by yusukebe** [comment]
Hi @ItsWendell !

The issue should be fixed and this PR is good! I've left some comments.

**Comment by ItsWendell** [comment]
@yusukebe appreciate it, I've cleaned up the tests and resolved your comments!

**Review by yusukebe — APPROVED** [review]
LGTM!

**Pull request merged**